Adding Inkplate 6 docs (#778)
OttoWinter/esphomedocs: components/display/Inkplate.rst

Learn more at `our website <https://inkplate.io/>`__

.. figure:: images/Inkplate.jpg
    :align: center
    :width: 75.0%

    Inkplate 6

.. code-block:: yaml

    # Example minimal configuration entry

    mcp23017:
      - id: mcp23017_hub
        address: 0x20

    display:
    - platform: inkplate
      id: inkplate_display
      greyscale: false
      partial_updating: false
      update_interval: 60s

      ckv_pin: 32
      sph_pin: 33
      gmod_pin:
        mcp23017: mcp23017_hub
        number: 1
      gpio0_enable_pin:
        mcp23017: mcp23017_hub
        number: 8
      oe_pin:
        mcp23017: mcp23017_hub
        number: 0
      spv_pin:
        mcp23017: mcp23017_hub
        number: 2
      powerup_pin:
        mcp23017: mcp23017_hub
        number: 4
      wakeup_pin:
        mcp23017: mcp23017_hub
        number: 3
      vcom_pin:
        mcp23017: mcp23017_hub
        number: 5

.. warning::

    When using the Inkplate epaper module, the GPIO pin numbers above *cannot be changed* as they are
    hardwired within the module/PCB.

Configuration variables
***********************

> 
> - **name** (**Required**, string): The name of the switch.
> - **greyscale**: Makes the screen display 3 bit colors.
> - **partial_updating**: Makes the screen update partialy, which is faster, but leaves burnin. 
> - **lambda** (*Optional*, :ref:`lambda <config-lambda>`): The lambda to use for rendering the content on the display.
>   See :ref:`display-engine` for more information.
> - **update_interval** (*Optional*, :ref:`config-time`): The interval to re-draw the screen. Defaults to ``5s``.
> - **pages** (*Optional*, list): Show pages instead of a single lambda. See :ref:`display-pages`.
> - **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
> 

Reviewer: ```suggestion

- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
- **greyscale** (*Optional*, boolean): Makes the screen display 3 bit colors. Defaults to ``False``
- **partial_updating** (*Optional*, boolean): Makes the screen update partially, which is faster, but leaves burnin. Defaults to ``False``
- **full_update_every** (*Optional*, int): When partial updating is enabled, forces a full screen update after chosen number of updates. Defaults to ``10``
- **lambda** (*Optional*, :ref:`lambda <config-lambda>`): The lambda to use for rendering the content on the display.
  See :ref:`display-engine` for more information.
- **update_interval** (*Optional*, :ref:`config-time`): The interval to re-draw the screen. Defaults to ``5s``.
- **pages** (*Optional*, list): Show pages instead of a single lambda. See :ref:`display-pages`.

```

Inkplate 6
==========

.. seo::
    :description: Instructions for setting up Inkplate E-Paper displays in ESPHome.
    :image: Inkplate.jpg

All-in-one e-paper display  ``Inkplate 6``
Inkplate 6 is a powerful, Wi-Fi enabled ESP32 based six-inch e-paper display – recycled from a Kindle e-reader. Its main feature is simplicity.

> Learn more at `our website <https://inkplate.io/>`__

Reviewer 2: Please write `inkplate's` website instead. I don't want to give the impression that ESPHome is somehow directly related to inkplate (users viewing the docs won't look who made the docs PR)

Author: Done that :D

- **display_data_4_pin** (*Optional*, :ref:`config-pin`): The Data 4 pin for the Inkplate display.
  Defaults to GPIO23.
- **display_data_5_pin** (*Optional*, :ref:`config-pin`): The Data 5 pin for the Inkplate display.
  Defaults to GPIO25.
- **display_data_6_pin** (*Optional*, :ref:`config-pin`): The Data 6 pin for the Inkplate display.
  Defaults to GPIO26.
- **display_data_7_pin** (*Optional*, :ref:`config-pin`): The Data 7 pin for the Inkplate display.
  Defaults to GPIO27.

Complete example
****************

The following is a complete example YAML configuration that does a few things beyond the usual
Wi-Fi, API, and OTA configuration.

.. code-block:: yaml

    # Example configuration entry
    esphome:
      name: inkplate
      platform: ESP32
      board: esp-wrover-kit

    logger:

    wifi:
      ssid: <YOUR WIFI SSID>
      password: <YOUR WIFI PASSWORD>
      ap:
        ssid: Inkplate-AP
        password: '12345678'

    captive_portal:

    ota:

    api:

    switch:
      - platform: restart
        name: "Inkplate Reboot"
        id: reboot

      - platform: gpio
        id: battery_read_mosfet
        pin:
          mcp23017: mcp23017_hub

>           number: 9

Reviewer: The pin should be inverted to read the battery voltage
```suggestion
          number: 9
          inverted: true
```

      - platform: template
        name: "Inkplate Greyscale mode"
        lambda: return id(inkplate_display).get_greyscale();
        turn_on_action:
          - lambda: id(inkplate_display).set_greyscale(true);
        turn_off_action:
          - lambda: id(inkplate_display).set_greyscale(false);

      - platform: template
        name: "Inkplate Partial Updating"
        lambda: return id(inkplate_display).get_partial_updating();
        turn_on_action:
          - lambda: id(inkplate_display).set_partial_updating(true);
        turn_off_action:
          - lambda: id(inkplate_display).set_partial_updating(false);

    sensor:
      - platform: adc
        id: battery_voltage
        update_interval: never
        attenuation: 11db
        pin: 35
      - platform: template
        name: "Inkplate Battery Voltage"
        lambda: |-

>           id(battery_read_mosfet).turn_on();
>           delay(1);
>           id(battery_voltage).update();
>           int adc = id(battery_voltage).state;
>           id(battery_read_mosfet).turn_off();
>           return adc;

Reviewer: Not sure how I missed this being a float
```suggestion
          id(battery_read_mosfet).turn_on();
          delay(1);
          float adc = id(battery_voltage).sample();
          id(battery_read_mosfet).turn_off();
          return adc;
```
